cli/src/main.rs

```diff
@@ -1,8 +1,89 @@
+use db::PgPool;
+use tokio;
+use clap::{Parser, Subcommand};
+use db::sqlx;
+use std::fs;
+use std::io::Write;
+use uuid::Uuid; // Ensure you have the `uuid` crate in your Cargo.toml
+
+#[derive(Parser)]
+#[command(author, version, about, long_about = None)]
+struct KnowledgeArgs {
+    #[command(subcommand)]
+    command: Command,
+}
+
+#[derive(Subcommand)]
+enum Command {
+    Login,
+    Signup,
+    DisplayUsers,
+}
```

Comment:
Here we have clap subcommands. We used this 

`cargo run -- signup --username alice`

in the above example, _signup_ is a subcommand whereas `username` is an argument. I was thinking it'd be nice to call an explicit "signup" command to signify the action you're trying to take, and then collect the arguments by prompting the user:

```
cargo run -- signup
enter a username:
seif
User ID: c750689f-572f-4b0d-b541-621085a74a51, Username: seif
```

An alternate approach would be to collect arguments along the subcommand:

```
cargo run -- signup --username alice
User ID: c750689f-572f-4b0d-b541-621085a74a51, Username: alice
```

OR we can just simplify this heavily and assume the program only runs signup and no other commands
```
cargo run --username alice
User ID: c750689f-572f-4b0d-b541-621085a74a51, Username: alice
```

_but_ the commands were nice for allowing us to check other things, like all the users in the DB (through the `display-users` subcommand for ex.)
```
cargo run -- display-users
User ID: c750689f-572f-4b0d-b541-621085a74a51, Username: alice
User ID: ac849baa-3cb1-40ef-a464-92460fa47617, Username: seif
```

Reply:
This comment is PERFECT 💯 

I love that not only did you explain what _AND_ why, but you gave alternatives too! 💜 

I like what you have now! The "interactive" sub-command style

But! If its not crazy complicated I think it would be really cool to support the `--username` style too!
Do you think it would work where if you don't give the `--username` option it prompts for it interactively?